src/Build/BuildCheck/Acquisition/BuildCheckAcquisitionModule.cs

```diff
@@ -42,17 +42,17 @@ public List<BuildAnalyzerFactory> CreateBuildAnalyzerFactories(AnalyzerAcquisiti
             assembly = Assembly.LoadFrom(analyzerAcquisitionData.AssemblyPath);
 #endif
 
-            IEnumerable<Type> analyzerTypes = assembly.GetExportedTypes().Where(t => typeof(BuildAnalyzer).IsAssignableFrom(t));
+            IEnumerable<Type> analyzerCandidates = assembly.GetExportedTypes();
 
-            foreach (Type analyzerType in analyzerTypes)
+            foreach (Type analyzerCandidate in analyzerCandidates)
             {
-                if (Activator.CreateInstance(analyzerType) is BuildAnalyzer instance)
+                if (Activator.CreateInstance(analyzerCandidate) is BuildAnalyzer instance)
                 {
                     analyzersFactories.Add(() => instance);
                 }
                 else
                 {
```

Comment:
Btw. how it can happen that the type satisfies `IsAssignableFrom`, but it cannot be `is` casted?